### PR TITLE
【翻译完成】20190705 Do you really need microservices?

### PR DESCRIPTION
**Pull request by Alex1996a**
重新 pr

### CODE REVIEW

**Comment by magichan** [comment]
#1298

File: translated/tech/20190705-Do-you-really-need-microservices?.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# 你真的需要微服务吗？
+
+## 前言
+我们已经[设计和构建](https://www.browserlondon.com/services/design-development/)了十多年的软件，大部分时间我们一直在使用优秀的 Symfony 框架来实现这一目标。 Symfony 是一个传统的单体 PHP 构件集，受 Java Spring 的启发，我们发现它非常适合[企业 web 应用程序](https://www.browserlondon.com/case-study/insights/)和[数字产品](https://www.browserlondon.com/case-study/twine/)的快速开发，而这些正是我们主要经济来源。
+
+然而，去年发布的 Symfony 4 代表了该框架的重点逐渐变化; 这变化体现在其远离单体架构和向[微服务](https://en.wikipedia.org/wiki/Microservices)靠拢，这种变化背后的方法论在过去几年中越来越受欢迎。
+
+为了说明这一转变，新版本在默认情况下使用了微内核（micro by default), Symfony 组织大力宣传其新的微内核设计，声称与 Symfony 3 相比，编写应用程序所需的代码减少了70%。
```
Comment (magichan) [inline translated/tech/20190705-Do-you-really-need-microservices?.md]: > 新版本在默认情况下使用了微内核（micro by default)

我个人还是不认同这句话的翻译，

> To illustrate this shift, the new release was billed as being ‘’micro by default’, 

 was billed as ：被称为，被标榜为

能否解释一下翻译的理由。

其他的没有什么问题。

Comment (Alex1996a) [reply translated/tech/20190705-Do-you-really-need-microservices?.md]: > > 新版本在默认情况下使用了微内核（micro by default)
> 
> 我个人还是不认同这句话的翻译，
> 
> > To illustrate this shift, the new release was billed as being ‘’micro by default’,
> 
> was billed as ：被称为，被标榜为
> 
> 能否解释一下翻译的理由。
> 
> 其他的没有什么问题。

查阅了symfony 3 有关 mirco by default 相关的资料,
其中 mirco by default 的描述如下：
> New applications created with Symfony 4 are based on a micro-kernel and contain 70% less code and files than new Symfony 3 apps! We worked tirelessly to optimize Symfony and remove any non-essential dependencies. The result is the leanest Symfony version ever created and the smallest footprint across all PHP frameworks.
参考资料地址： https://symfony.com/4 
个人认为， mirco by default 和 based on a micro-kernel 这个等价，

并且，使用主动句式读起来会更通顺一点。
1、为了说明这一转变，新版本在默认情况下使用了微内核（micro by default)
2、为了说明这一转变，新版本被标榜为默认微服务

Comment (magichan) [reply translated/tech/20190705-Do-you-really-need-microservices?.md]: OK.

**Pull request merged**